### PR TITLE
Allow certain users to Create Company

### PR DESCRIPTION
**Pull request by marrwin**

### CODE REVIEW

File: codershq/companies/forms.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#         username = self.cleaned_data["username"]
+
+#         try:
+#             User.objects.get(username=username)
+#         except User.DoesNotExist:
+#             return username
+
+#         raise ValidationError(self.error_messages["duplicate_username"])
+
+class CompanyCreationForm(ModelForm):
```
Comment (ralsuwaidi) [inline codershq/companies/forms.py:26]: better not to leave the comments above, select one of the classes and remove the other when you are confident it works

File: codershq/companies/views.py [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
+                return render(request, self.template_name, {'form':CompanyCreationForm})
+            else:
+                return render(request, 'companies/not_allowed.html')
+        else:
+            return render(request, 'pages/welcome.html')
+
+    def post(self, request, *args, **kwargs):
+        form = CompanyCreationForm(request.POST, request.FILES)
+        if form.is_valid():
+            name = request.POST['name']
```
Comment (ralsuwaidi) [inline codershq/companies/views.py:38]: isnt there a better way to save the form? I rememebr the form saving itself without you explicitly taking all the values and creating a new object from them.

File: codershq/users/models.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ class User(AbstractUser):
     # users can be part of a team
     # cannot delete team if a user is part of that team
     teams = models.ForeignKey('Team', on_delete=models.PROTECT, null=True, blank=True)
+    allowed_to_create_company = models.BooleanField(default=False)
```
Comment (ralsuwaidi) [inline codershq/users/models.py:32]: change this to something shorter, like `is_company` for example

**Review by ralsuwaidi — CHANGES_REQUESTED** [review]

**Comment by ralsuwaidi** [comment]
overall its a very decent job. Just a few changes and the update view.

**Review by gitanimous — COMMENTED** [review]
I agree with rashed using something like `is_company` instead of `allowed_to_create_company` would be better, great work man 👍

**Comment by ralsuwaidi** [comment]
closed PR due to changes not being updated for a while

**Pull request closed without merging**